dask_planner/src/sql.rs

```diff
@@ -160,6 +161,9 @@ impl ContextProvider for DaskSQLContext {
         for schema in self.schemas.values() {
             for (fun_name, function) in &schema.functions {
                 if fun_name.eq(name) {
+                    if function.lock().unwrap().aggregation.eq(&true) {
+                        return None;
+                    }
```

Comment:
This mutex needs to be manually `unlocked` since the RAII guard will still be in scope at this point. https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.lock I think it best to refactor this snippet like this